src/Microsoft.Health.Dicom.Functions.UnitTests/Management/PurgeOrchestrationInstancesHistoryTests.cs

```diff
@@ -0,0 +1,132 @@
+        {
+            using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+
+            _durableOrchestrationClientMock
+                .ListInstancesAsync(
+                    Arg.Is<OrchestrationStatusQueryCondition>(givenCondition => IsCondition(givenCondition, null)),
+                    Arg.Is<CancellationToken>(cancellationTokenSource.Token))
+                .Returns(Task.FromResult(new OrchestrationStatusQueryResult { DurableOrchestrationState = Array.Empty<DurableOrchestrationStatus>() }));
+
+            await _purgeOrchestrationInstancesHistory.Run(_timer, _durableOrchestrationClientMock, _logger, cancellationTokenSource.Token);
+
+            await _durableOrchestrationClientMock
+                .DidNotReceiveWithAnyArgs()
+                .PurgeInstanceHistoryAsync(Arg.Any<string>());
+        }
+
+        [Fact]
+        public async Task GivenValidInput_PurgeCompletedDurableFunctionsHistory_ShouldPurgeMultiPageListOfOrchestrationsAsync()
+        {
+            using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+
+            List<DurableOrchestrationStatus> durableOrchestrationStatuses = new List<DurableOrchestrationStatus>()
+            {
+                new DurableOrchestrationStatus{ InstanceId = "1" },
+                new DurableOrchestrationStatus{ InstanceId = "2" },
+                new DurableOrchestrationStatus{ InstanceId = "3" }
+            };
+
+            OrchestrationStatusQueryResult orchestrationStatusQueryResult = new OrchestrationStatusQueryResult();
+            orchestrationStatusQueryResult.DurableOrchestrationState = durableOrchestrationStatuses;
+            orchestrationStatusQueryResult.ContinuationToken = "fake_cancellation_token";
+
+            var collector = new List<string>();
+
+            _durableOrchestrationClientMock
+                .ListInstancesAsync(
+                    Arg.Is<OrchestrationStatusQueryCondition>(
+                            givenCondition => IsCondition(givenCondition, null)),
+                    Arg.Is<CancellationToken>(cancellationTokenSource.Token))
+                // First page, should have 3 orchestrations, then change continuation token.
+                .Returns(_ =>
+                    {
+                        orchestrationStatusQueryResult.ContinuationToken = null;
+                        Assert.True(orchestrationStatusQueryResult.DurableOrchestrationState.SequenceEqual(durableOrchestrationStatuses));
+                        return Task.FromResult(orchestrationStatusQueryResult);
+                    });
+
+            _durableOrchestrationClientMock
+                .ListInstancesAsync(
+                    Arg.Is<OrchestrationStatusQueryCondition>(
+                            givenCondition => IsCondition(givenCondition, orchestrationStatusQueryResult.ContinuationToken)),
```

Comment:
nit: too large of an indentation. Some changes may help save space. For example, I think you could use a shorter parameter name like `x` here as it's fairly obvious what the argument is. Furthermore, I would copy the expected continuation token inline instead of referencing this value